zrcs/zrc-2.md

```diff
@@ -0,0 +1,412 @@
+
+
+#### 9. AuthorizeOperator
+
+```ocaml
+(* @dev: Make an address an operator of the caller.                           *)
+(* @param operator: Address to be set as operator. Cannot be calling address. *)
+transition AuthorizeOperator(operator: ByStr20)
+```
+
+|        | Name       | Type      | Description                    |
+| ------ | ---------- | --------- | ------------------------------ |
+| @param | `operator` | `ByStr20` | Address to be set as operator. Cannot be calling address. |
+
+|           | Name                       | Description                             | Event Parameters                                                                                                                                                                                                               |
+| --------- | -------------------------- | --------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| eventName | `AuthorizeOperatorSuccess` | Authorizing is successful.     | `operator`: `ByStr20`. |
+| eventName | `Error`                    | Authorizing is not successful. | - emit `CodeNotAuthorised` if the transition is called by a user who is not the token holder.                                                                                                      |
+
+
+#### 10. RevokeOperator
+
+```ocaml
+(* @dev: Revoke an address from being an operator of the caller. *)
+(* @param operator:         Address to be unset as operator.     *)
+transition RevokeOperator(operator: ByStr20)
+```
+
+|        | Name       | Type      | Description                      |
+| ------ | ---------- | --------- | -------------------------------- |
+| @param | `operator` | `ByStr20` | Address to be unset as operator. |
+
+|           | Name                       | Description                             | Event Parameters                                                                                                                                                                                                               |
+| --------- | -------------------------- | --------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| eventName | `RevokeOperatorSuccess` | Revoking is successful.     | `operator`: `ByStr20`. |
+| eventName | `Error`                    | Revoking is not successful. | - emit `CodeNotAuthorised` if the transition is called by a user who is not the token holder.                                                                                                      |
```

Comment:
Same comment as for `AuthorizeOperator`.

Reply:
@vaivaswatha if the operator is not found then it cannot be revoked.